lib/ui/hooks.dart

```diff
@@ -27,40 +30,117 @@ void _updateWindowMetrics(
   double/*!*/ systemGestureInsetBottom,
   double/*!*/ systemGestureInsetLeft,
 ) {
-  window
-    .._devicePixelRatio = devicePixelRatio
-    .._physicalSize = Size(width, height)
-    .._physicalDepth = depth
-    .._viewPadding = WindowPadding._(
-        top: viewPaddingTop,
-        right: viewPaddingRight,
-        bottom: viewPaddingBottom,
-        left: viewPaddingLeft)
-    .._viewInsets = WindowPadding._(
-        top: viewInsetTop,
-        right: viewInsetRight,
-        bottom: viewInsetBottom,
-        left: viewInsetLeft)
-    .._padding = WindowPadding._(
-        top: math.max(0.0, viewPaddingTop - viewInsetTop),
-        right: math.max(0.0, viewPaddingRight - viewInsetRight),
-        bottom: math.max(0.0, viewPaddingBottom - viewInsetBottom),
-        left: math.max(0.0, viewPaddingLeft - viewInsetLeft))
-    .._systemGestureInsets = WindowPadding._(
-        top: math.max(0.0, systemGestureInsetTop),
-        right: math.max(0.0, systemGestureInsetRight),
-        bottom: math.max(0.0, systemGestureInsetBottom),
-        left: math.max(0.0, systemGestureInsetLeft));
-  _invoke(window.onMetricsChanged, window._onMetricsChangedZone);
+  assert(PlatformDispatcher.instance._screens[screenId] != null);
```

Comment:
here and in other entry points, i think we should move most of this logic to a private member of the class on which most of the fields are being updated. it's weird to have this top-level private poking directly into the privates of some other object.